src/density/occupation_matrix.hpp

```diff
@@ -45,18 +47,38 @@ class Occupation_matrix : public Hubbard_matrix {
 
     void reduce()
     {
+        if (!ctx_.hubbard_correction())
+            return;
+
         /* global reduction over k points */
-        for (int ia = 0; ia < ctx_.unit_cell().num_atoms(); ia++) {
-            if (ctx_.unit_cell().atom(ia).type().hubbard_correction()) {
-                ctx_.comm_k().allreduce(this->local(ia).at(memory_t::host), static_cast<int>(this->local(ia).size()));
+        for (int at_lvl = 0; at_lvl < (int)this->local_.size(); at_lvl++) {
+            const int ia     = atomic_orbitals_[at_lvl].first;
+            auto const& atom = ctx_.unit_cell().atom(ia);
+            if (atom.type().lo_descriptor_hub(atomic_orbitals_[at_lvl].second).use_for_calculation()) {
+                ctx_.comm().allreduce(this->local(at_lvl).at(memory_t::host),
+                                      static_cast<int>(this->local(at_lvl).size()));
             }
         }
+
+        // we must reduce occ_mtrx_T_ not nonlocal (it non zero only after symmetrization)
+        for (auto& T : this->occ_mtrx_T_) {
+            ctx_.comm().allreduce(T.second.at(memory_t::host), static_cast<int>(T.second.size()));
+        }
+
+        // std::cout  << "Before : " << this->nonlocal(0)(0, 0, 0) << std::endl;
```

Comment:
these are debugging statements I forgot to remove. We can most probably just delete them.